### PR TITLE
[SIG-2754] Add configuration for showing the extra questionary

### PR DESCRIPTION
**Pull request by aditudorache**
This PR
- adds a new property in the application configuration: `hasExtraProps`
- when this property si set to false, no extra questionary wil be presented to the user.
- this property will be used to switch off the extra questionaries for the Wesp implementation

Note that the specific configuration of the build repositories for Weesp and Amsterdam wil be done after this PR is Approved

### CODE REVIEW

File: environment.conf.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
       "gemeentenaam": "amsterdam"
     }
   },
+  "hasExtraProps": true,
```
Comment (janjaap) [inline environment.conf.json:57]: I think the name of this prop is too generic. Should be something like `showVulaanControls`. At least use naming that has already been used in the application.

File: src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import configuration from 'shared/services/configuration/configuration';
+import vullaan from '../wizard-step-2-vulaan';
+
+describe('wizard-step-2-vullaan', () => {
```
Comment (janjaap) [inline src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js:4]: One 'l' too many

File: src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import configuration from 'shared/services/configuration/configuration';
+import vullaan from '../wizard-step-2-vulaan';
```
Comment (janjaap) [inline src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js:2]: One 'l' too many

File: src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('wizard-step-2-vullaan', () => {
+  it('should return controls when hasExtraProps is true', () => {
+    configuration.hasExtraProps = true;
+    expect(vullaan.formFactory({ category: "afval" }).controls).not.toEqual({ });
```
Comment (janjaap) [inline src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js:7]: Should be no space in `{ }`

File: src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  });
+
+  it('should return empty controls when hasExtraProps is false', () => {
+    configuration.hasExtraProps = false;
+    expect(vullaan.formFactory({ category: "afval" }).controls).toEqual({ });
```
Comment (janjaap) [inline src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js:12]: Should be no space in `{ }`

File: src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  });
+
+  it('should return empty controls when hasExtraProps is true but no category is found', () => {
+    configuration.hasExtraProps = false;
+    expect(vullaan.formFactory({ category: "not-existing-category" }).controls).toEqual({ });
```
Comment (janjaap) [inline src/signals/incident/definitions/__tests__/wizard-step-2-vulaan.test.js:17]: Should be no space in `{ }`

File: src/signals/incident/definitions/wizard-step-2-vulaan.js [hunk 7]
```diff
@@ -15,6 +16,9 @@ export default {
   previousButtonClass: 'action startagain',
   formAction: 'UPDATE_INCIDENT',
   formFactory: ({ category }) => {
+    const noExtraProps = { controls: {} };
+    if(!configuration.hasExtraProps) return noExtraProps;
```
Comment (janjaap) [inline src/signals/incident/definitions/wizard-step-2-vulaan.js:20]: Missing space after `if`

File: environment.conf.json (hunk 1 above)
Comment (aditudorache) [reply environment.conf.json:57]: shure thing!

File: src/signals/incident/definitions/wizard-step-2-vulaan.js [hunk 8]
```diff
@@ -17,7 +17,7 @@ export default {
   formAction: 'UPDATE_INCIDENT',
   formFactory: ({ category }) => {
     const noExtraProps = { controls: {} };
-    if(!configuration.hasExtraProps) return noExtraProps;
+    if (!configuration.showVulaanControls) return noExtraProps;
```
Comment (janjaap) [inline src/signals/incident/definitions/wizard-step-2-vulaan.js:20]: The configuration object with throw an exception when a prop isn't there. Might be better to wrap this in a try/catch

Comment (aditudorache) [reply src/signals/incident/definitions/wizard-step-2-vulaan.js:20]: Added extra check but be aware that this extra check is not don for any of the configuration object properties...

Comment (janjaap) [reply src/signals/incident/definitions/wizard-step-2-vulaan.js]: You're right. In the case of querying for props of the `configuration` object, we can actually do two things:

__try/catch__
```jsx
let nestedProp;
try {
  nestedProp = configuration.map.options.nestedProp;
} catch (e) {
 //
}
```

__conditional chaining__
```
const nestedProp = configuration?.map?.options?.nestedProp;
```

In your case, conditional chaining would be fine.

**Comment by bartjkdp** [comment]
This change breaks our deployments that do not contain the `showVulAan` configuration:

image

**Comment by janjaap** [comment]
@bartjkdp In what way? Not having that prop in the configuration should gracefully degrade because of the conditional chaining in `wizard-step-2-vulaan`

**Comment by bartjkdp** [comment]
@janjaap not having this key in the configuration throws an exception that is not caught. I created an issue for this 👍

**Comment by janjaap** [comment]
@bartjkdp Absolutely right; can't believe we missed that. PR: https://github.com/Amsterdam/signals-frontend/pull/959

**Comment by bartjkdp** [comment]
@janjaap no problem :) thanks fix fixing this so quickly!

**Pull request merged**